includes/class-base-widget.php

```diff
@@ -518,12 +610,33 @@ public function enqueue_scripts() {
 		$suffix = SCRIPT_DEBUG ? '' : '.min';
 
 		wp_enqueue_style( 'font-awesome', '//maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css', [], '4.5.0' );
-		wp_enqueue_style( 'wpcw-admin', \Contact_Widgets::$assets_url . "css/admin{$suffix}.css", [ 'font-awesome' ], Plugin::$version );
-		wp_enqueue_script( 'wpcw-admin', \Contact_Widgets::$assets_url . "js/admin{$suffix}.js", [ 'jquery' ], Plugin::$version, true );
+		wp_enqueue_style( 'jquery-timepicker', Plugin::$assets_url . "css/jquery.timepicker{$suffix}.css", [ 'font-awesome' ], '1.11.9' );
```

Comment:
@EvanHerman Since the timepicker is used exclusively by the hours widget, it shouldn't go in the base widget. Instead, we should use method inheritance inside `class-hours.php`. Additionally, we should move `font-awesome` to `class-social.php`.

Another thing to try would be to add an arg to the `enqueue_scripts` method that would allow localize script data to be passed to from every inherited method back to the parent base.

The code below is untested, but it's an example of what I mean:

```php
/* class-hours.php */

public function enqueue_scripts( $localize_script_data = [] ) {

	$localize_script_data = [
		'time_format' => get_option( 'time_format' ),
	];

	parent::enqueue_scripts( $localize_script_data );

	// Stuff here the hours widget needs to enqueue, like jquery-timepicker

}
```

```php
/* class-base-widget.php */

public function enqueue_scripts( $localize_script_data = [] ) {

	// Stuff here every widget needs to enqueue, like wpcw-admin

	static $data = [];

	$data = array_merge_recursive( $data, $localize_script_data );

	if ( $data ) {

		wp_localize_script( 'wpcw-admin', 'wpcw_admin', $data );

	}

}
```